pkgs/by-name/bm/bms-to-osu/package.nix

```diff
@@ -0,0 +1,54 @@
+{
+  lib,
+  fetchFromGitHub,
+  buildDotnetModule,
+  dotnetCorePackages,
+  fetchurl,
+  ncurses,
+  ffmpeg,
+}:
+
+let
+  pname = "bms-to-osu";
+  version = "2.5-unstable-2024-12-20"; # 2.5 crashes at runtime due to missing kernel32.dll
+in
+buildDotnetModule {
+  inherit pname version;
+
+  src = fetchFromGitHub {
+    owner = "QingQiz";
+    repo = "BmsToOsu";
+    rev = "67b28435fd9ad1dd1b396efdb8a26e974c8e280a"; # tag = "v${version}";
+    hash = "sha256-0k+qwauyHdlcCOUl58K985HHciJI/4/5U4YogDvbZSA=";
+  };
+
+  patches = [
+    # fix cannot find ffmpeg in unix systems
+    (fetchurl {
+      url = "https://github.com/QingQiz/BmsToOsu/commit/715e3ff61d9ff22463dc4672410702348d690cb1.patch";
+      hash = "sha256-kI0DUQqqhYrdtVBsvcucDOG2IEgVgRQbw76OnjARU4c=";
+    })
+  ];
+
+  makeWrapperArgs = [
+    "--suffix"
+    "PATH"
+    ":"
+    "${ffmpeg}/bin"
+  ];
+
+  nugetDeps = ./deps.json;
+  projectFile = "BmsToOsu.sln";
+  executables = [ "BmsToOsu" ];
+  dotnet-runtime = dotnetCorePackages.runtime_6_0;
```

Comment:
How feasible is it to use a newer runtime?

Reply:
```nix
  postPatch = ''
    sed -i 's|<TargetFramework>net6.0</|<TargetFramework>net8.0</|' BmsToOsu/BmsToOsu.csproj
  '';
```
This seems to work. Do I need to switch to .NET 8.0?

Reply:
Sounds good to me. Maybe upstream will agree with updating it so we don't need to patch it.